examples/user_guide/Reactive_Expressions.ipynb

```diff
@@ -0,0 +1,258 @@
+   "source": [
+    "Now we will create a reactive function using `param.bind`. We start by declaring a function that adds two parameters and then bind parameters `a` and `b` to that function:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "c0a4c64e-ee64-405d-bace-605551b34234",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def add(a, b):\n",
+    "    return a+b\n",
+    "\n",
+    "p = Parameters()\n",
+    "\n",
+    "reactive_add = param.bind(add, p.param.a, p.param.b)\n",
+    "\n",
+    "reactive_add"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "9094768c-5a39-448e-9721-e6253a2a55cb",
+   "metadata": {},
+   "source": [
+    "You can see that this reactive function renders the output and when we update one of the inputs to the function the output will reflect the updated result."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "f7cd00ac-02a2-4e0c-bc98-7ce9739ed55b",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "p.a = 5"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "9a0e5412-b13f-42aa-bb21-4bf6f289510b",
+   "metadata": {},
+   "source": [
+    "We can also call the reactive function to return the current result:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "1e026152-1772-4437-8165-dd9788edd1fb",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "reactive_add()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "03d62757-8b3f-4d67-b1a9-5759194408a2",
+   "metadata": {},
+   "source": [
+    "Note that if you only partially bind the required arguments the function will not be automatically evaluated:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "88870bf0-1d5e-42e9-9c59-b538070ad701",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "param.bind(add, p.param.a)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "5a774c07-63d7-4f40-be59-ce52242bdbd6",
+   "metadata": {},
+   "source": [
+    "## Reactive Expressions\n",
+    "\n",
+    "While reactive functions are very useful and allow writing arbitrary logic they still require writing the actual functions which result in more indirection and less readability. With a `reactive` expression you can wrap any object or parameter value and apply operations on it, as if you are working with the actual object. In other words, it acts as a proxy of the underlying value and supports (almost) all operations that can be performed with the original object.\n",
+    "\n",
+    "### Using Parameters\n",
+    "\n",
+    "As an example we can create reactive proxies for the `a` and `b` parameters and add them together:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "243afd62-2f10-4dd3-a264-4d9a895c3088",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "p = Parameters()\n",
+    "\n",
+    "p.param.a.reactive() + p.param.b.reactive()"
```

Comment:
Its "clever". But there is a lot for users to learn here. Just the `.param` namespace is hard for users. They either don't understand it. Or they don't use all its features. Its what I see on Discourse.

Reply:
Why do you add `.reactive` to parameters and `param.bind`. Why don't you just use `param.reactive` and parameters or `.bind` as input?

I'm asking because below you showcase `param.reactive` and now there are two ways to use `reactive` and not just one.

Reply:
> Its "clever". But there is a lot for users to learn here. Just the .param namespace is hard for users. They either don't understand it. Or they don't use all its features. Its what I see on Discourse. Now there is .reactive on top.

True but imho that's a documentation issue, even the best API needs to be properly taught and referenced in the documentation. I believe that the effort made in this PR to push functionality (originally in hvPlot!) down the stack into Param will force us to better introduce Param in ways that are appropriate for Panel and HoloViews users.